src/main/java/org/jabref/gui/search/GlobalSearchBar.java

```diff
@@ -307,7 +308,12 @@ public void performSearch() {
             informUserAboutInvalidSearchQuery();
             return;
         }
-        this.stateManager.addSearchHistory(searchField.textProperty().get());
+        searchBoxFocused.bind(searchField.focusedProperty());
```

Comment:
this whole block needs to be outside the performSearch.
Currently you add a new listener for every search. But we need to have only one listener. See e.g. line 212